### PR TITLE
generateReductionSpec.py: Don't split arms/spectrographs.

### PR DESCRIPTION
**Pull request by SogoMineo**
generateReductionSpec.py previously generated separate YAML blocks
for different arms. Now YAML blocks are merged if they are mergeable.

### CODE REVIEW

File: python/pfs/pipe2d/generateReductionSpec.py [hunk 1]
```diff
@@ -765,3 +766,192 @@ def nameYamlMapping(name: str, mapping: Dict[str, Any]) -> Dict[str, Any]:
...
+def _mergeCalibBlocks_merge(objects: Iterable[Any]) -> Any:
+    """Merge calibBlocks.
+
+    Parameters
+    ----------
+    objects : `Iterable[Any]`
+        A list, each element being any constituent part of a calibBlock.
+        (including a complete calibBlock itself).
+        All elements must be mergeable to each other.
+        Otherwise ValueError is raised.
```
Comment (PaulPrice) [inline python/pfs/pipe2d/generateReductionSpec.py:860]: I suggest putting this under a `Raises` heading.

File: python/pfs/pipe2d/generateReductionSpec.py [hunk 2]
```diff
@@ -765,3 +766,192 @@ def nameYamlMapping(name: str, mapping: Dict[str, Any]) -> Dict[str, Any]:
     newMapping = {"name": name}
     newMapping.update((key, value) for key, value in mapping.items() if key != "name")
     return newMapping
+
+
+def mergeCalibBlocks(blocks: Iterable[Any]) -> List[Any]:
+    """Find mergeable calibBlocks in ``blocks`` and merge them.
+    Two calibBlocks are "mergeable"
```
Comment (PaulPrice) [inline python/pfs/pipe2d/generateReductionSpec.py:773]: Leave a blank line between the short header description and the details.

**Review by PaulPrice — APPROVED** [review]
Just some trivial comments.

File: python/pfs/pipe2d/generateReductionSpec.py (hunk 1 above)
Comment (SogoMineo) [reply python/pfs/pipe2d/generateReductionSpec.py:860]: I deleted the phrase. In the course of writing this code, there was a time when this `ValueError` was actually caught by a caller. At that time ValueError had to be a part of API. But it no longer has to be in the current code.

**Pull request merged**